feature(astrojs/cloudflare): add support for `splitted` SSR bundles (#7464)
withastro/astro: packages/integrations/cloudflare/README.md

For most projects the adapter default of `advanced` will be sufficient; the `dist` folder will contain your compiled project. Switching to directory mode allows you to use [pages plugins](https://developers.cloudflare.com/pages/platform/functions/plugins/) such as [Sentry](https://developers.cloudflare.com/pages/platform/functions/plugins/sentry/) or write custom code to enable logging.

In directory mode, the adapter will compile the client side part of your app the same way by default, but moves the worker script into a `functions` folder in the project root. In this case, the adapter will only ever place a `[[path]].js` in that folder, allowing you to add additional plugins and pages middleware which can be checked into version control.

> With the build configuration `split: true`, the adapter instead compiles a seperate bundle for each page which is used by cloudflare routing. Be aware that this will overwrite any files in the `functions` folder with the same name emitted by Astro. It is recommended to name your custom files, e.g. Pages Plugins & Middleware, distinctly to avoid accidental overwrite. Additionally the adapter will not empty the `functions` folder, so you might need to clean-up the folder if you remove pages. This is a difference to the `dist` folder which is always cleaned up between builds, which is necessery to not remove custom files in the `functions` folder. For now the adapter will bundle the middleware into the each page, and will not make use of [Cloudflare Pages Middleware](https://developers.cloudflare.com/pages/platform/functions/middleware/). We are working to improve the adapter, so you can set `build.excludeMiddleware: true`, to get proper supported output.

Author: @sarah11918 thanks, some technical details are not quite right, so I tried to update that with you suggestions in mind.

Also the build configuration option is not added by me, it already exists in core.

For most projects the adapter default of `advanced` will be sufficient; the `dist` folder will contain your compiled project. Switching to directory mode allows you to use [pages plugins](https://developers.cloudflare.com/pages/platform/functions/plugins/) such as [Sentry](https://developers.cloudflare.com/pages/platform/functions/plugins/sentry/) or write custom code to enable logging.

In directory mode, the adapter will compile the client side part of your app the same way by default, but moves the worker script into a `functions` folder in the project root. In this case, the adapter will only ever place a `[[path]].js` in that folder, allowing you to add additional plugins and pages middleware which can be checked into version control.

> Cloudflare documentation contains more information about [writing custom functions](https://developers.cloudflare.com/pages/platform/functions/).
> 
> With the build configuration `split: true`, the adapter instead compiles a separate bundle for each page. This option requires some manual maintenance of the `functions` folder. Files emitted by Astro will overwrite existing `functions` files with identical names, so you must choose unique file names for each file you manually add. Additionally the adapter will never empty the `functions` folder of outdated files, so you must clean-up the folder manually when you remove pages.
> 
> Additionally, this adapter does not support using [Cloudflare Pages Middleware](https://developers.cloudflare.com/pages/platform/functions/middleware/). Astro will bundle the [Astro Middleware](https://docs.astro.build/en/guides/middleware/) into each page.

Reviewer: ```suggestion
With the build configuration `split: true`, the adapter instead compiles a separate bundle for each page. This option requires some manual maintenance of the `functions` folder. Files emitted by Astro will overwrite existing `functions` files with identical names, so you must choose unique file names for each file you manually add. Additionally, the adapter will never empty the `functions` folder of outdated files, so you must clean up the folder manually when you remove pages.

Note that this adapter does not support using [Cloudflare Pages Middleware](https://developers.cloudflare.com/pages/platform/functions/middleware/). Astro will bundle the [Astro middleware](https://docs.astro.build/en/guides/middleware/) into each page.
```
Tiny nits to fix, including I just didn't like having "Additionally," twice in a row like that!  :sweat_smile: (Not wrong, just trying to go for some sentence variety. Either one could be changed!)

Also, the link to writing Cloudflare custom functions now feels a little out of place? Can we get away without it entirely?

Otherwise, happy to sign off on this form of the document!

Author: Makes sense :)

I think we can get away without the link, because the Middleware link also links to their docs. People writing custom functions, will probably know the Cloudflare Docs anyways.

For most projects the adapter default of `advanced` will be sufficient; the `dist` folder will contain your compiled project. Switching to directory mode allows you to use [pages plugins](https://developers.cloudflare.com/pages/platform/functions/plugins/) such as [Sentry](https://developers.cloudflare.com/pages/platform/functions/plugins/sentry/) or write custom code to enable logging.

> In directory mode the adapter will compile the client side part of your app depending on the new `build.split` setting. If `split` is disabled the adapter compiles everything in the same way, but moves the worker script into a `functions` folder in the project root. The adapter will only ever place a `[[path]].js` in that folder, allowing you to add additional plugins and pages middleware which can be checked into version control.

Reviewer: ```suggestion
In directory mode, the adapter will compile the client side part of your app the same way by default, but moves the worker script into a `functions` folder in the project root. In this case, the adapter will only ever place a `[[path]].js` in that folder, allowing you to add additional plugins and pages middleware which can be checked into version control.
```
P. S. We don't refer to things as new, because that "gets old" quickly!  :smile: 

And, *especially* if this is intended to be experimental, but even if not, I would tend to instead just describe the above behaviour as "default" (since by default, `split` will not be true). The paragraph below can describe what happens if someone has actively configured `split: true`.

> `Experimental:` if `split` is enabled the adapter compiles for each pages a seperate bundle, which is used by cloudflare routing. Cloudflare documentation contains more information about [writing custom functions](https://developers.cloudflare.com/pages/platform/functions/).

Reviewer: ```suggestion
With the build configuration `split: true`, the adapter instead compiles a seperate bundle for each page which is used by cloudflare routing. Cloudflare documentation contains more information about [writing custom functions](https://developers.cloudflare.com/pages/platform/functions/).
```
I think this might be clear enough without needing to overload the first paragraph with information about `build.split`, since people might not even set that configuration?

If this is determined to be experimental (not just "new, and I hope it doesn't break" but rather behind a flag), then ping Docs again to determine the appropriate syntax for marking as experimental!

For most projects the adapter default of `advanced` will be sufficient; the `dist` folder will contain your compiled project. Switching to directory mode allows you to use [pages plugins](https://developers.cloudflare.com/pages/platform/functions/plugins/) such as [Sentry](https://developers.cloudflare.com/pages/platform/functions/plugins/sentry/) or write custom code to enable logging.

In directory mode the adapter will compile the client side part of your app depending on the new `build.split` setting. If `split` is disabled the adapter compiles everything in the same way, but moves the worker script into a `functions` folder in the project root. The adapter will only ever place a `[[path]].js` in that folder, allowing you to add additional plugins and pages middleware which can be checked into version control.

`Experimental:` if `split` is enabled the adapter compiles for each pages a seperate bundle, which is used by cloudflare routing. Cloudflare documentation contains more information about [writing custom functions](https://developers.cloudflare.com/pages/platform/functions/).

> 

Author: this prob need feedback of @withastro/maintainers-docs!
How do you guys normally document experimental stuff in adapters?

With the build configuration `split: true`, the adapter instead compiles a seperate bundle for each page which is used by cloudflare routing. Be aware that this will overwrite any files in the `functions` folder with the same name emitted by Astro. It is recommended to name your custom files, e.g. Pages Plugins & Middleware, distinctly to avoid accidental overwrite. Additionally the adapter will not empty the `functions` folder, so you might need to clean-up the folder if you remove pages. This is a difference to the `dist` folder which is always cleaned up between builds, which is necessery to not remove custom files in the `functions` folder. For now the adapter will bundle the middleware into the each page, and will not make use of [Cloudflare Pages Middleware](https://developers.cloudflare.com/pages/platform/functions/middleware/). We are working to improve the adapter, so you can set `build.excludeMiddleware: true`, to get proper supported output.

> Cloudflare documentation contains more information about [writing custom functions](https://developers.cloudflare.com/pages/platform/functions/).

Author: Added some more information, so users know what to expect, regarding features.
can you check this, again? @sarah11918
